GuessTheMovie/ImageStorage.h

```diff
@@ -53,7 +53,23 @@ struct ImageStorage
 			images.push_back(dummy_image());
 		}
 		return images;
-	}
+	}*/
+	std::vector<Image> get_folder_images(const std::string& path)
+	{
+		
+		std::vector<Image> image_vector;
+		boost::filesystem::directory_iterator it(path);
+		while (it != boost::filesystem::directory_iterator())
+		{
+			if ((it->path().extension() == ".jpg") || (it->path().extension() == ".jpeg") 
+				|| (it->path().extension() == ".png") || (it->path().extension() == ".gif")
+				|| (it->path().extension() ==".JPG")||(it->path().extension() == ".JPEG")
+				|| (it->path().extension() ==".PNG")||((it->path().extension() == ".GIF")))
```

Comment:
Can't you simplify this using the <algorithm> library? For example you could have a

const std::vector<std::string> image_extensions = { ".jpg", ".JPG", ... };

and see if it->path().extension() is in image_extensions